repose-aggregator/src/docs/asciidoc/architecture/filter-chain.adoc

```diff
@@ -0,0 +1,106 @@
+= Filter Chain
+
+The *Repose* filter chain is the work horse of the system, and is responsible for nearly all enrichment that *Repose* provides.
+It's based on and uses the same contract as the https://docs.oracle.com/javaee/6/api/javax/servlet/FilterChain.html[JEE Filter Chain].
+As the name implies it is a chain of filters that are followed in order.
+A request goes through them from the first, one after another, until the last is reached when the request which has possibly been mutated along the way is sent on to the origin service.
+When the response comes back from the origin service it works it's way back up the chain in the opposite order.
+
+The *Repose* filter chain provides a number of enhancements over the standard filter chain contract that gives a lot of power and extra value.
+
+== Dynamic Filtering
+
+Typically a filter chain is static and unchanging.
+Your request will pass through every single filter everytime regardless of whether you want it to or not.
+The *Repose* filter chain allows you to determine if a filter should be run while the request is processing.
+
+=== URI Regex ===
+
+[NOTE]
+====
+This method of determining if a filter should be run is *DEPRECATED* and will be removed in *Repose* 10.
+====
+
+You can decide whether or not a filter should run based on it's URI using a https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html[regular expression].
```

Comment:
I think technically we only match the `uri-regex` against the path from the URI right now. That might be worth noting so that folks aren't trying to match the protocol, host, or port portion of the URI.